cairo/ethereum/cancun/vm/memory.cairo

```diff
@@ -0,0 +1,202 @@
+// SPDX-License-Identifier: MIT
+
+from starkware.cairo.common.alloc import alloc
+from starkware.cairo.common.bool import FALSE, TRUE
+from starkware.cairo.common.default_dict import default_dict_new, default_dict_finalize
+from starkware.cairo.common.dict import DictAccess, dict_read, dict_write
+from starkware.cairo.common.memset import memset
+from starkware.cairo.common.memcpy import memcpy
+from starkware.cairo.lang.compiler.lib.registers import get_fp_and_pc
+from starkware.cairo.common.math import assert_le, assert_lt
+from starkware.cairo.common.math_cmp import is_le, is_not_zero
+
+from ethereum_types.bytes import Bytes, BytesStruct, Bytearray, BytearrayStruct, Bytes1DictAccess
+from ethereum_types.numeric import U256
+
+// @notice Write bytes to memory at a given position.
+// @dev assumption: start_position < 2**128
+// @dev assumption: value.len < 2**128
```

Comment:
I think that assumption in the context of cairo means that it's not checked if it's true or false. also I don't see where there is any constraint on `value.len`

Reply:
we don't want start_position + value.len to overflow a uint128